torchvision/prototype/datasets/utils/_internal.py

```diff
@@ -40,10 +39,9 @@ def read_mat(buffer: BinaryIO, **kwargs: Any) -> Any:
     except ImportError as error:
         raise ModuleNotFoundError("Package `scipy` is required to be installed to read .mat files.") from error
 
-    if isinstance(buffer, StreamWrapper):
-        buffer = buffer.file_obj
```

Comment:
This is no longer needed for quite some time, since `StreamWrapper` no correctly exposes all methods of the `file_obj` and thus can be used directly in `sio.loadmat`.